From aaa122251b174f4cc3cdc3acca1b6b980ac6da60 Mon Sep 17 00:00:00 2001
From: Zoltan Gera <zoltan.gera@qt.io>
Date: Wed, 7 Jan 2026 15:03:45 +0200
Subject: [PATCH] Do not resolve symbolic links in generate_jinja.py

Special Windows builds using symbolic links and substitute drives for
tricking path length limitations can be broken by resolving symbolic
links. Functionality canonizing directory path should keep away from
real_path conversions.
---
 data/generators/generate_jinja.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/data/generators/generate_jinja.py b/data/generators/generate_jinja.py
index ba7e78861..f527b76b5 100755
--- a/data/generators/generate_jinja.py
+++ b/data/generators/generate_jinja.py
@@ -24,9 +24,9 @@ def check_grammar_dirs(args):
         if not os.path.isdir(gr_dir):
             print(f'Search folder {gr_dir} does not exist.')
             sys.exit(1)
-        args.grammar_dir[i] = os.path.realpath(gr_dir)
+        args.grammar_dir[i] = os.path.abspath(gr_dir)
     for grammar_xml in args.grammar_xmls:
-        folder = os.path.dirname(os.path.realpath(grammar_xml))
+        folder = os.path.dirname(os.path.abspath(grammar_xml))
         if folder not in args.grammar_dir:
             args.grammar_dir.append(folder)
 
@@ -69,7 +69,7 @@ def frontend():
             sys.exit(1)
     if args.output_dir is None:
         args.output_dir = '.'
-    args.output_dir = os.path.realpath(args.output_dir)
+    args.output_dir = os.path.abspath(args.output_dir)
     if not os.path.isdir(args.output_dir):
         print('Destination folder does not exist.')
         sys.exit(1)
-- 
GitLab

